### PR TITLE
change soil moisture initialization logic for FATES configurations

### PR DESCRIPTION
**Pull request by ckoven**
This PR changes the logic for soil moisture initialization to initialize with wetter soils (75% of saturated water content, as opposed to 15% of absolute water content) for all FATES configurations.  The rationale for this is that in FATES-nocomp simulations, @JessicaNeedham was finding very high initial mortality rates in some seasonal tropical forest regions, and she traced it back to the initial soil moisture killing off plants before the ecosystem could get established.  More info at https://github.com/NGEET/fates/issues/994.

### Description of changes
This soil moisture value was already in use for FATES-Hydro, this PF just updates it to be used for all FATES configurations.  I've run it and the code does the intended thing, which is to make soils considerably wetter across the world during the initial period from a cold-start spinup. 

The thing I am curious about is whether there are any longer-term implications to the soil moisture, in particular in the permafrost region.  I haven't tested this for a long period yet, but I wonder how much the initial soil moisture gets locked in for permafrost soil layers, and thus how sensitive they are to initial conditions even after long spinup? @swensosc and @dlawrenncar probably know the answer to this.

### Specific notes

Contributors other than yourself, if any: @JessicaNeedham

~~CTSM~~ FATES Issues Fixed (include github issue #): https://github.com/NGEET/fates/issues/994

Are answers expected to change (and if so in what way)? Yes, in all non-hydro FATES configurations (the change to initial soil moisture is already the case for FATES-hydro and so wouldn't change anything there)

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any:

@JessicaNeedham has done more complete testing in E3SM-FATES to show that it fixes the initial high mortality bias. I have run this code in CLM-FATES and it does increase soil moisture as intended.  I have not run it through any test suites.

### CODE REVIEW

**Comment by wwieder** [comment]
We're wondering we can / should keep the HLM and FATES initialization the same to avoid confusion for users?  This seems like it warrants a more scientific-focused discussion with both FATES and HLM teams?

**Comment by ekluzek** [comment]
We also wondered if this shouldn't be a namelist item of some sort to make it more obvious what the starting state is being set to.

**Comment by glemieux** [comment]
> We're wondering we can / should keep the HLM and FATES initialization the same to avoid confusion for users? This seems like it warrants a more scientific-focused discussion with both FATES and HLM teams?

Per the ctsm software meeting discussion today, we agreed to bring in this PR as is and continue the broader science-focused discussion in a new repo issue/discussion.  I'll use this PR as a vehicle to update fates to the latest API updates as well.

**Comment by glemieux** [comment]
Testing the `fates` suite on Cheyenne is underway.  I'll follow this up with testing on izumi and the `aux_clm` suite tomorrow assuming the fates results look as expected.

**Comment by glemieux** [comment]
Almost all `fates` tests are passing as expected, with DIFFs on all tests except the `Hydro` test mods, which are B4B. 

That said, one of our longer, higher resolution exact tests (`ERS_Lm13.f10_f10_mg37.I2000Clm50Fates.cheyenne_gnu`) is barely failing to complete due to hitting the wallclock.  Previous tests show that this test was already skirting the limit and with more cohorts surviving the computation for fates has increased, pushing the test beyond the wallclock limit.  For the short term, I'm going to slightly increase the wallclock to see if the result passes as expected.  For the longterm, I think it might be good to revisit this particular test's overlap with others to see if we should change it since it's so long.

**Comment by glemieux** [comment]
> For the short term, I'm going to slightly increase the wallclock to see if the result passes as expected. 

The test ran as expected taking 42.6 minutes.

File: cime_config/testdefs/testlist_clm.xml
Comment (rgknox) [inline cime_config/testdefs/testlist_clm.xml:1]: nice

**Review by rgknox — APPROVED** [review]
I can't think of anything else this needs.  Thanks for also updating the FATES tag and associated parameter file.

**Comment by glemieux** [comment]
`aux_clm` regression testing on cheyenne is complete.  All fates tests returned DIFFs as expected since we are updating to the latest the fates science tag.  

One tests passed although it was expected to fail: `FUNITCTSM_P1x1.f10_f10_mg37.I2000Clm50Sp.cheyenne_intel`

There was one unexpected failure, `FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.cheyenne_intel`, that returned as `failed to initialize` for the `SHAREDLIB_BUILD` step.  I'm not sure what happened here.  Have you ever seen this @ekluzek?

Folder on cheyenne: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_dev121-aux`
Izumi testing coming soon.

**Comment by ekluzek** [comment]
@glemieux odd that the UNIT tests worked for you, but it had been failing for @billsacks. Maybe CISL fixed something on the system? Or it's a user dependent thing?

On the other issue it's failing in this set of commands...

`. /glade/scratch/glemieux/ctsm-tests/tests_dev121-aux/FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.cheyenne_intel.GC.dev121-aux_int/.env_mach_specific.sh; module unload python; module load conda; conda activate ctsm_pylib && python3 /glade/u/home/glemieux/ctsm/tools/modify_input_files/fsurdat_modifier /glade/scratch/glemieux/ctsm-tests/tests_dev121-aux/FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.cheyenne_intel.GC.dev121-aux_int/modify_fsurdat.cfg`

Try running each of those commands separately on the command line and see what fails. You might also just

``` bash
./py_env_create
conda activate ctsm_pylib
cd python
make test
```

and see what happens...

**Comment by glemieux** [comment]
Thanks @ekluzek.  It looks like everything worked as expected:
```
(ctsm_pylib) ✔  ~/ctsm/python [modinitsoilwater {ckoven/modinitsoilwater } ↓·1|⚑ 12] 
11:46 $ make test
python3 ./run_ctsm_py_tests  --unit
.....................................................................................................................................................................
----------------------------------------------------------------------
Ran 165 tests in 0.988s

OK
python3 ./run_ctsm_py_tests  --sys
...............Setting resource.RLIMIT_STACK to -1 from (307200000, -1)
Done converting /glade/scratch/glemieux/tmp_otnm7it/scrip.nc
.
----------------------------------------------------------------------
Ran 16 tests in 12.257s

OK
```
Maybe this was a glitch of some sort?  Should I try and manually run `./case.build` for that test?

**Comment by ekluzek** [comment]
Possibly what happened was that you hadn't run py_env_create before so you needed to run it by hand? Could that be the base? 

We have a long list of things that happen in that line in the script. Possibly the script should do them separately, so you know which one fails.

**Comment by glemieux** [comment]
Oh, I didn't realize we needed to run `py_env_create` prior to `run_sys_tests -s aux_clm`.  Loading the `ctsm_pylib` env and then re-running `./case.build` in the test dir worked.  I was able to then submit and that test ran successfully and returned B4B.

**Comment by ekluzek** [comment]
Yep, this is a new thing that you need to do. We possibly need more documentation and advice given about the need to do it.

**Comment by glemieux** [comment]
I'm running into an issue with the nag compiler izumi tests.  I'm getting the following error:
```
Error: /home/glemieux/ctsm/src/fates/main/FatesHistoryInterfaceMod.F90, line 2478: Too many continuation lines
```

We've definitely increased the number of variables in the `association` statement since the last fates science tag update.  For comparison I was able to create a baseline tag for fates using `fates-sci.1.61.0_api.25.0.0-ctsm5.1.dev120`, which is the current tag that ctsm is pointing to in the externals.  I've never seen this before.  Have you @rgknox, @ekluzek?

**Comment by ekluzek** [comment]
@glemieux yes I saw this with the CN matrix work. There is a limit for nag on how many lines you can have in those continue statements. What we had to do for CN matrix was to have more than one associate statement that you nest. It's annoying but not that much different and just adds a couple end associate statements at the subroutine close.

**Comment by glemieux** [comment]
I'll be testing https://github.com/NGEET/fates/pull/1009 and updating the externals pointer to the integrated for fix the above issue prior to continuing with this PR.

**Comment by billsacks** [comment]
Regarding the FUNIT tests: Brian Vanderwende in CISL made a system change that should get them working again. He said:

> Jim is right that 4.9.0 added a libxml2 dependency that has been problematic on Cheyenne (as there is no "libxml2.so" on the compute image - though there is "libxml2.so.2"). We account for this after 4.9.0 and so using 4.9.1 should work fine as Jim mentions.
>
> That said, I've put a hack in that should hopefully fix the issue for 4.9.0 as well. Let me know how it goes for you.

So I have closed #1972. @glemieux I opened a PR with some changes that should be merged now that this is resolved: #1978 . Can you either (a) cherry-pick the single commit on that branch into the branch you're working on here, or (b) give me the go-ahead to merge #1978 when it won't interfere with your work?

**Comment by glemieux** [comment]
Thanks for the heads up @billsacks.  I'll cherry-pick your commit into this PR.

**Comment by glemieux** [comment]
Testing on Cheyenne is complete.  All expected non-fates test pass B4B for the `aux_clm` suite.  All expected tests pass bfb in the `fates` suite.

Folders on Cheyenne:
```
/glade/u/home/glemieux/scratch/ctsm-tests/tests_dev121-aux-1
/glade/u/home/glemieux/scratch/ctsm-tests/tests_dev121-fates-1
```

**Comment by glemieux** [comment]
Testing on izumi is complete.  As expected all non-fates tests are b4b for the `aux_clm` tests.  All non-hydro tests are not b4b as expected for the `fates` suite/ 

Folders on Izumi:
```
/scratch/cluster/glemieux/ctsm-tests/tests_dev121-aux_clm
/scratch/cluster/glemieux/ctsm-tests/tests_dev121-fates-1
```

**Comment by glemieux** [comment]
`fates` suite regression tests are complete on Cheyenne.   All expected tests PASS.  The non-hydro testmods are the only B4B tests as expected.  

Folder on cheyenne: 
```
/glade/u/home/glemieux/scratch/ctsm-tests/tests_dev120-121-fates-65-3
```
@ekluzek this should be good to go now.

**Review by ekluzek — APPROVED** [review]

**Comment by dlawrenncar** [comment]
Is this new logic only applicable to FATES?  It seems like we could
potentially use the same logic in big leaf CTSM to avoid a divergence
between the two configurations.  Something to discuss.

**Comment by ekluzek** [comment]
@dlawrenncar yes it's only for FATES and is considered a temporary solution. This just allows them to run for now.

We've had some discussion about better long term solutions, but they will require some study. And likely someone identified to work on this. See comments near the top of this issue. We thought this should be discussed at an up and coming CTSM science meeting.

**Pull request merged**